stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/adapter/viewholder/ChannelViewHolder.kt

```diff
@@ -53,9 +54,50 @@ public class ChannelViewHolder @JvmOverloads constructor(
     private val optionsMenuWidth
         get() = menuItemWidth * optionsCount
 
+    private lateinit var channel: Channel
+
+    init {
+        binding.apply {
+            root.setOnClickListener {
+                channelClickListener.onClick(channel)
+            }
+            root.setOnLongClickListener {
+                channelLongClickListener.onClick(channel)
+                true
+            }
+
+            itemBackgroundView.apply {
+                moreOptionsImageView.setOnClickListener {
+                    channelMoreOptionsListener.onClick(channel)
+                    swipeListener.onSwipeCanceled(this@ChannelViewHolder, absoluteAdapterPosition)
+                }
+                deleteImageView.setOnClickListener {
+                    channelDeleteListener.onClick(channel)
+                }
+            }
+
+            itemForegroundView.apply {
+                avatarView.setOnClickListener {
+                    when {
+                        channel.isDirectMessaging() -> userClickListener.onClick(currentUser)
+                        else -> channelClickListener.onClick(channel)
+                    }
+                }
+
+                applyStyle(style)
```

Comment:
Good, better than `styleApplied` flag

Reply:
Which was also not set anywhere previously, just to make it worse 😉